### ISSUE TITLE
Setting VM Hardware to version 15 prevents the install to complete successfully  

### ISSUE THREAD

**Issue by smarunich** [cptmorgan-rh/ocp4-aio-vsphere-upi-lab#6]
The current playbook does upgrade VM hardware to the version 15, that has impact on the openshift provisioning within the low-end lab hardware, the specs provided in the environment section. However, keeping VM hardware to default version (13 within 6.7/7.0) allows provisioning to complete.

Is there a particular reason for VM hardware to be set to 15?

## Expected Behavior

The playbook has to complete successfully, the cluster should be available for use.

## Actual Behavior

Creating VMs for OCP 4.7.1 with VM hardware as 15, the issue i have faced described in upstream: https://github.com/openshift/okd/issues/430

openshift wait-for install-complete does not complete successfully

In nutshell, switch to VM hardware 15 prevents the cluster to successfully initialize, tested on 4.7.0 and 4.7.1, however the default VM version which is 13 allows to provision the cluster and no issues have been observed.

the workaround was to:
```--- a/roles/deploy_ova/tasks/main.yml
+++ b/roles/deploy_ova/tasks/main.yml
@@ -37,6 +37,6 @@
     command: "govc vm.change -vm {{ vcenter.folder_absolute_path }}/{{ vcenter.template_name }} -e=disk.EnableUUID=TRUE" # noqa 204
     when: skip_ova is not defined

-  - name: Upgrade VM version on the template
-    command: "govc vm.upgrade -vm {{ vcenter.folder_absolute_path }}/{{ vcenter.template_name }} -version=15" # noqa 204
-    when: skip_ova is not defined
+#  - name: Upgrade VM version on the template
+#    command: "govc vm.upgrade -vm {{ vcenter.folder_absolute_path }}/{{ vcenter.template_name }} -version=15" # noqa 204
+#    when: skip_ova is not defined
```

## Environment

OS: Red Hat Enterprise Linux release 8.3 (Ootpa)
Ansible Version: 2.9.17
OpenShift Version: 4.7.1
ESXi Version: 7.0.1b17551050

```
> smarunich@master:~$ govc host.info zoo01.lab.sm
> Name:              zoo01.lab.sm
>   Path:            /lab/host/zoo/zoo01.lab.sm
>   Manufacturer:    MSI
>   Logical CPUs:    12 CPUs @ 2400MHz
>   Processor type:  Intel(R) Xeon(R) CPU E5-2620 v3 @ 2.40GHz
>   CPU usage:       4961 MHz (17.2%)
>   Memory:          130968MB
>   Memory usage:    48408 MB (37.0%)
>   Boot time:       2021-03-12 21:16:08.807 +0000 UTC
>   State:           connected
> smarunich@master:~$ govc host.info zoo02.lab.sm
> Name:              zoo02.lab.sm
>   Path:            /lab/host/zoo/zoo02.lab.sm
>   Manufacturer:    Intel(R) Client Systems
>   Logical CPUs:    8 CPUs @ 2304MHz
>   Processor type:  Intel(R) Core(TM) i5-8259U CPU @ 2.30GHz
>   CPU usage:       1002 MHz (5.4%)
>   Memory:          32628MB
>   Memory usage:    18278 MB (56.0%)
>   Boot time:       2021-03-12 21:00:37.395 +0000 UTC
>   State:           connected
> smarunich@master:~$ govc host.info zoo03.lab.sm
> Name:              zoo03.lab.sm
>   Path:            /lab/host/zoo/zoo03.lab.sm
>   Manufacturer:    Intel(R) Client Systems
>   Logical CPUs:    8 CPUs @ 2304MHz
>   Processor type:  Intel(R) Core(TM) i5-8259U CPU @ 2.30GHz
>   CPU usage:       413 MHz (2.2%)
>   Memory:          32632MB
>   Memory usage:    17010 MB (52.1%)
>   Boot time:       2021-03-12 21:00:55.456 +0000 UTC
>   State:           connected
> smarunich@master:~$ govc host.info zoo04.lab.sm
> Name:              zoo04.lab.sm
>   Path:            /lab/host/zoo/zoo04.lab.sm
>   Manufacturer:    Intel(R) Client Systems
>   Logical CPUs:    12 CPUs @ 1607MHz
>   Processor type:  Intel(R) Core(TM) i7-10710U CPU @ 1.10GHz
>   CPU usage:       3205 MHz (16.6%)
>   Memory:          65242MB
>   Memory usage:    35267 MB (54.1%)
>   Boot time:       2021-03-12 21:20:43.426 +0000 UTC
>   State:           connected
```

## Commands used to run ansible-playbook
ansible-playbook deploy-aio-lab.yml

## Ansible Playbook Output

the issue i am facing described in the upstream: https://github.com/openshift/okd/issues/430

## Ansible group_vars
```
config:
  provider: vsphere
  base_domain: ocp.lab.sm
  cluster_name: cluster1
  cluster_domain: cluster1.ocp.lab.sm
  fips: false
  installer_ssh_key: "{{ lookup('file', '~/.ssh/id_rsa.pub') }}"
  pull_secret: "{{ lookup('file', '~/pull-secret.json') }}"
  worker_prefix: cluster1-aio-worker
vcenter:
  ip: 192.168.0.15
  datastore: minas-LUN-3
  network: zoo-comp-pg-vmnetwork
  service_account_username: 
  service_account_password: 
  admin_username: 
  admin_password: 
  datacenter: lab
  cluster: zoo
  folder_absolute_path:
  vm_power_state: poweredon
  template_name: cluster1-aio-rhcos-vmware
download:
  channel: stable
  version: 4.7
  govc: https://github.com/vmware/govmomi/releases/download/v0.24.0/govc_linux_amd64.gz
  fcct: https://github.com/coreos/fcct/releases/download/v0.10.0/fcct-x86_64-unknown-linux-gnu
coredns_vm:
  name: cluster1-aio-coredns
  ipaddr: 192.168.1.10
  upstream_dns: 192.168.0.100
lb_vm:
  name: cluster1-aio-haproxy
  ipaddr: 192.168.1.11
bootstrap_vm:
  name: cluster1-bootstrap
  ipaddr: 192.168.1.16
master_vms:
  - { name: "cluster1-aio-master-0", ipaddr: "192.168.1.13", cpus: "4", memory_mb: "16384", disk_size_gb: "120"}
  - { name: "cluster1-aio-master-1", ipaddr: "192.168.1.14", cpus: "4", memory_mb: "16384", disk_size_gb: "120"}
  - { name: "cluster1-aio-master-2", ipaddr: "192.168.1.15", cpus: "4", memory_mb: "16384", disk_size_gb: "120"}
worker_vms:
  - { name: "cluster1-aio-worker-0", ipaddr: "192.168.1.17", cpus: "2", memory_mb: "8192", disk_size_gb: "120"}
  - { name: "cluster1-aio-worker-1", ipaddr: "192.168.1.18", cpus: "2", memory_mb: "8192", disk_size_gb: "120"}
  - { name: "cluster1-aio-worker-2", ipaddr: "192.168.1.19", cpus: "2", memory_mb: "8192", disk_size_gb: "120"}
static_ip:
  gateway: 192.168.0.1
  netmask: 255.255.248.0
  network_interface_name: ens192
ntp:
  custom: true
  ntp_server_list:
    - 192.168.0.1
    - 192.168.0.100
 ```

**Comment by cptmorgan-rh** [comment]
> Is there a particular reason for VM hardware to be set to 15?

It is set to V15 for the VMware CSI Plugin which it requires for the CSI plugin to work.

I have switched to OVN from SDN because I was having issues with SDN. I will remove the option for now until the issue is resolved.

**Comment by cptmorgan-rh** [comment]
This was resolved in commit d221df84fc
